deploy/quickstart/quick_start.py

```diff
@@ -0,0 +1,211 @@
+        # than one default-required field like this:
+        replace_in_file(capabilities_config, "\"isRequired\": true", "\"isRequired\": false")
+
+        print("To start, run:")
+        print("> " + start_script_file)
+
+    else:
+        print("Sorry, not yet implemented! Please raise this issue at https://github.com/DataBiosphere/job-manager/issues")
+        exit(1)
+
+
+def quick_start_dsub():
+    print("Sorry, not yet implemented! Please raise this issue at https://github.com/DataBiosphere/job-manager/issues")
+
+
+def download_file(version, file, to_path):
+    abs_to_path = path.abspath(to_path)
+    from_url = "https://raw.githubusercontent.com/DataBiosphere/job-manager/" + version + "/" + file
+    urllib.request.urlretrieve(from_url, abs_to_path)
+    return abs_to_path
+
+
+def replace_in_file(file, to_replace, replace_with):
+    with open(file, "rt") as fin:
+        temp = "temp_out.txt"
+        with open("temp_out.txt", "wt") as fout:
+            for line in fin:
+                fout.write(line.replace(to_replace, replace_with))
+        call(["mv", "-f", temp, file])
+
+
+def write_string(file, lines):
+    with open(file, "wt") as fout:
+        for line in lines:
+            fout.writelines(line + "\n")
+
+
+def pull_docker(name, version):
+    docker_image = name + ":" + version
+    print("Pulling docker image " + docker_image)
+    call(["docker", "pull", docker_image], stdout=subprocess.DEVNULL)
+    return docker_image
+
+
+def request_input_from_list(message, options, default):
+    valid = False
+    provided = default
+    if not default in options:
+        printable_option_list = options.copy()
+        all_options = options.copy()
+        all_options = all_options + default
+    else:
+        all_options = options
+        printable_option_list = options.copy()
+        printable_option_list.remove(default)
+
+    printed_options_list = "(" + "/".join(str(e) for e in all_options) + ")"
+
+    while not valid:
+        provided = input(">> " + message + " " + printed_options_list + " [ " + default + " ] > ") or default
+        if provided in all_options:
+            valid = True
+        else:
+            print("Invalid option. Expected one of: " + printed_options_list + " but got: " + provided)
+            valid = False
+    print("Using: " + provided)
+    print()
+    return provided
+
+
+def make_or_replace_or_reuse_directory(path_to_make):
+    valid = False
+    time_to_exit = False
+    while not time_to_exit:
+        if path.isdir(path_to_make):
+            print("Directory " + path_to_make + " already exists. I can...")
+            print("1. Default: Re-use the existing directory which might leave behind some old files")
+            print("2. Delete the entire directory and all of its contents - and then remake it, completely empty")
+            provided = input("Replace or re-use? (1/2) [ 1 ] > ") or "1"
+            if provided is 1:
```

Comment:
Note,  this will run into a dead loop since provided is a `string` and it is checking if a `string` == `int` here. To resolve it, either use `provided = int(input(...))` or use `if provided is str(1):` or `if provided is "1"`

Reply:
Besides, usually we use == when comparing values and `is` when comparing identities, when comparing ints (or immutable types, like str in general), you might want the former to avoid some traps. 

There's an "optimization" which lets small ints to be compared with `is`, but that will be problematic for larger cases:
```
>>> a = 19999999998
>>> b = 19999999997 + 1
>>> a is b
False
>>> a == b
True
```
For boolean comparisions `is` is always more pythonic tho (`if not`, `if True`...).